internal/impl/docker.go

```diff
@@ -109,57 +144,42 @@ func buildImage(ctx context.Context, specs *imageSpecs) error {
 	if err != nil {
 		return err
 	}
-	if err := dockerfileTmpl.Execute(dockerFile, specs.goInstall); err != nil {
+	if err := dockerfileTmpl.Execute(dockerFile, spec.goInstall); err != nil {
 		dockerFile.Close()
 		return err
 	}
 	if err := dockerFile.Close(); err != nil {
 		return err
 	}
-	return dockerBuild(ctx, workDir, specs.name)
-}
-
-// Use docker-cli to build the docker image.
-func dockerBuild(ctx context.Context, buildContext, tag string) error {
-	c := exec.CommandContext(ctx, "docker", "build", buildContext, "-t", tag)
-	c.Stdout = os.Stdout
-	c.Stderr = os.Stderr
-	return c.Run()
+	return dockerBuild(ctx, workDir, spec.tag)
 }
 
-// uploadImage upload image appImage to docker hub.
-func uploadImage(ctx context.Context, appImage string) error {
-	fmt.Fprintf(os.Stderr, greenText(), fmt.Sprintf("\nUploading image %s...", appImage))
-
-	c := exec.CommandContext(ctx, "docker", "push", appImage)
-	c.Stdout = os.Stdout
+// dockerBuild builds a docker image given a directory and an image tag.
+func dockerBuild(ctx context.Context, dir, tag string) error {
+	fmt.Fprintln(os.Stderr, "Building with tag:", tag)
+	c := exec.CommandContext(ctx, "docker", "build", dir, "-t", tag)
+	c.Stdout = os.Stderr
```

Comment:
any reason why is not os.Stdout? Is it because we print stuff only at os.Stderr? c.Stdout = os.Stderr here and in a few other places in the change.